### PR TITLE
Add webhook failure counting

### PR DESCRIPTION
**Pull request by thinkst-marco**
## Proposed changes

Add webhook failure counting

If a webhook fails 5 (or more) times, then it is disabled. The token owner can still manual re-enable the token on the management page.

## Types of changes

What types of changes does your code introduce to this repository?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] Lint and unit tests pass locally with my changes (if applicable)
- [X] I have run pre-commit (`pre-commit` in the repo)
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: canarytokens/constants.py [hunk 1]
```diff
@@ -25,3 +25,6 @@
 CANARYTOKEN_LENGTH = 25  # equivalent to 128-bit id
 
 CANARY_PDF_TEMPLATE_OFFSET: int = 793
+
+# Maximum number of alert failures before a mechanism is disabled
+MAX_ALERT_FAILURES = 5
```
Comment (jayjb) [inline canarytokens/constants.py:30]: Do we not want to maybe make this configurable? With the default being 5.

File: canarytokens/canarydrop.py [hunk 2]
```diff
@@ -467,3 +469,27 @@ def format_triggered_details_of_history_page(self) -> dict[str, Any]:
         """
 
         return self.triggered_details.serialize_for_v2(readable_time_format=True)
+
+    def clear_alert_failures(self) -> None:
+        if self.alert_failure_count:
+            self.alert_failure_count = 0
+            queries.save_canarydrop(self)
+
+    def record_alert_failure(self) -> None:
```
Comment (jayjb) [inline canarytokens/canarydrop.py:478]: Should this be
```
def record_alert_failure(self) -> int:
```

File: canarytokens/canarydrop.py [hunk 3]
```diff
@@ -467,3 +469,27 @@ def format_triggered_details_of_history_page(self) -> dict[str, Any]:
...
+
+    def has_too_many_alert_failures(self) -> bool:
+        if (
+            self.alert_failure_count is None
+            or self.alert_failure_count < MAX_ALERT_FAILURES
+        ):
+            return False
+        return True
+
+    def disable_alert_webhook(self) -> bool:
```
Comment (jayjb) [inline canarytokens/canarydrop.py:493]: And this:
```
def disable_alert_webhook(self) -> None:
```

**Review by jayjb — CHANGES_REQUESTED** [review]

File: canarytokens/canarydrop.py (hunk 2 above)
Comment (thinkst-marco) [reply canarytokens/canarydrop.py]: Good call, moved to Settings

File: canarytokens/constants.py (hunk 1 above)
Comment (thinkst-marco) [reply canarytokens/constants.py]: Good call, moved to Settings.

**Review by jayjb — APPROVED** [review]

**Pull request merged**